engine/pokemon/summary_screen.asm

```diff
@@ -0,0 +1,1030 @@
+	; Presumes the low byte of hBGMapAddress is 00
+	ld a, 32
+	ldh [hBGMapAddress], a
+	call ApplyTilemapInVBlank
+	call ApplyAttrmapInVBlank
+	xor a
+	ldh [hBGMapAddress], a
+
+	call UpdateSprites
+	call LoadFontsBattleExtra
+	call SummaryScreen_InitTiles
+
+	ldh a, [rSVBK]
+	push af
+	ld a, BANK(wOBPals1)
+	ldh [rSVBK], a
+	ld hl, .SummaryScreenObjPalettes
+	ld bc, 8 palettes
+	ld de, wOBPals1
+	rst CopyBytes
+	farcall ApplyOBPals
+	pop af
+	ldh [rSVBK], a
+
+	ld a, 1
+	ldh [hCGBPalUpdate], a
+
+	ld a, [wTempMonBox]
+	ld b, a
+	ld a, [wTempMonSlot]
+	ld c, a
+	farcall GetStorageBoxMon
+
+	ld hl, wSummaryScreenInterrupts
+	ld a, 143
+	ld [hli], a
+	ld [hli], a
+	ld bc, 2 * 15
+	ld a, -1
+	rst ByteFill
+	; Set up for HBlank
+
+	ld hl, hFunctionTarget
+	ld b, [hl]
+	ld a, LOW(LCDSummaryScreenHideWindow)
+	ld [hli], a
+	ld c, [hl]
+	push bc
+	ld a, HIGH(LCDSummaryScreenHideWindow)
+	ld [hl], a
+	ld hl, rIE
+	ld a, [hl]
+	push af
+	set LCD_STAT, [hl]
+	ld a, 100
+	ld [rLYC], a
+	; save old window position
+	ldh a, [hWX]
+	ld b, a
+	ldh a, [hWY]
+	ld c, a
+	push bc
+
+	; Load initial mon
+	xor a
+	ld [wSummaryScreenFlags], a
+	call SummaryScreen_InitMon
+	call DelayFrame
+	call SummaryScreen_LoadPage
+
+	; Enter loop
+	call SummaryScreenLoop
+	; Clean up
+
+	call ClearBGPalettes
+	call ClearTileMap
+
+	; restore old values
+	pop bc
+	ld a, b
+	ldh [hWX], a
+	ld a, c
+	ldh [hWY], a
+
+	pop af
+	ld [rIE], a
+
+	pop bc
+	ld hl, hFunctionTarget
+	ld [hl], b
+	inc hl
+	ld [hl], c
+
+	pop bc
+	ld a, b
+	ld [wJumptableIndex], a
+	ld a, c
+	ld [wSummaryScreenFlags], a
+	pop af
+	ld [wBoxAlignment], a
+	pop af
+	ldh [hMapAnims], a
+	ret
+
+.SummaryScreenObjPalettes:
+; pink
+	RGB 31, 31, 31
+	RGB 31, 15, 31
+	RGB 00, 00, 00
+	RGB 00, 00, 00 ; placeholder
+; blue
+	RGB 31, 31, 31
+	RGB 17, 31, 31
+	RGB 00, 00, 00
+	RGB 00, 00, 00 ; placeholder
+; green
+	RGB 31, 31, 31
+	RGB 17, 31, 00
+	RGB 00, 00, 00
+	RGB 00, 00, 00 ; placeholder
+; orange
+	RGB 31, 31, 31
+	RGB 30, 22, 12
+	RGB 00, 00, 00
+	RGB 00, 00, 00 ; placeholder
+; green info button
+	RGB 31, 31, 31
+	RGB 16, 24, 10
+	RGB 31, 31, 31
+	RGB 00, 00, 00 ; placeholder
+; unused
+	RGB 31, 31, 31
+	RGB 31, 31, 31 ; unused
+	RGB 31, 31, 31 ; unused
+	RGB 00, 00, 00 ; placeholder
+; arrow dark
+	RGB 31, 31, 31
+	RGB 31, 31, 31
+	RGB 31, 10, 06
+	RGB 00, 00, 00 ; placeholder
+; arrow light
+	RGB 31, 31, 31
+	RGB 31, 31, 31
+	RGB 31, 20, 20
+	RGB 00, 00, 00 ; placeholder
+
+SummaryScreen_InitTiles:
+	ld hl, GFX_Summary
+	ld de, vTiles2 tile $31
+	lb bc, BANK(GFX_Summary), 18
+	call DecompressRequest2bpp
+
+	ld a, 1
+	ldh [rVBK], a
+	ld de, TypeIconGFX
+	ld hl, vTiles3 tile $00
+	lb bc, BANK(TypeIconGFX), 19 * 4
+	call Request1bpp
+	xor a
+	ldh [rVBK], a
+
+	ld de, CategoryIconGFX
+	ld hl, vTiles2 tile $50
+	lb bc, BANK(CategoryIconGFX), 2 * NUM_CATEGORIES
+	call Request2bpp
+
+	ld hl, GFX_Summary_Sprites
+	ld de, vTiles0 tile $3C
+	lb bc, BANK(GFX_Summary_Sprites), 8 * 4
+	jp DecompressRequest2bpp
```

Comment:
Polished Crystal uses the `jmp` macro instead of the `jp` instruction. The former comes with an assertion that will warn if the jump can be optimized back to a `jr`.